src/containers/tutor-home-page/general-info-step/GeneralInfoStep.jsx

```diff
@@ -1,12 +1,136 @@
 import Box from '@mui/material/Box'
+import Typography from '@mui/material/Typography'
 
+import { useState, useEffect } from 'react'
+import { useSelector } from 'react-redux'
+import { useTranslation } from 'react-i18next'
+
+import generalInfo from '~/assets/img/tutor-home-page/become-tutor/general-info.svg'
+import AppTextField from '~/components/app-text-field/AppTextField'
+import AsyncAutocomplete from '~/components/async-autocomlete/AsyncAutocomplete'
 import { styles } from '~/containers/tutor-home-page/general-info-step/GeneralInfoStep.styles'
+import { LocationService } from '~/services/location-service'
+import { userService } from '~/services/user-service'
+import getEmptyArrayData from '~/utils/get-empty-array-data'
 
 const GeneralInfoStep = ({ btnsBox }) => {
+  const { t } = useTranslation()
+  const [firstName, setFirstName] = useState('')
+  const [lastName, setLastName] = useState('')
+  const [country, setCountry] = useState('')
+  const [city, setCity] = useState('')
+  const [textField, setTextField] = useState('')
+  const [charCount, setCharCount] = useState(0)
+
+  const countryTextFieldProps = {
+    label: t('common.labels.country')
+  }
+  const cityTextFieldProps = {
+    label: t('common.labels.city')
+  }
+  const store = useSelector((state) => state.appMain)
+
+  useEffect(() => {
+    const fetchUserData = async () => {
+      const userData = await userService.getUserById(
+        store.userId,
+        store.userRole
+      )
+      setFirstName(userData.data.firstName)
+      setLastName(userData.data.lastName)
+    }
+    fetchUserData()
+  }, [store.userId, store.userRole])
+
+  const handleTextFieldChange = (event) => {
+    const inputValue = event.target.value
+    if (inputValue.length <= 100) {
+      setTextField(inputValue)
+      setCharCount(inputValue.length)
+    }
+  }
+
+  const handleCountryChange = (value) => {
+    setCountry(value)
+    setCity(null)
+  }
+
+  const handleCityChange = (value) => {
+    setCity(value)
+  }
```

Comment:
Try to combine hooks in several custom hooks. maybe `useUserName`,  `useUserLoactaion` and `useText`. Yo can combine not only `useState` but also `useEffect`s for corresponding states.

Reply:
I've create useUserName custom hook and tests for it!
I 've decided don't create useUserLoactaion hook, because we already haveLocationService ('~/services/location-service) and use it in our component!